File: ova player/app/src/main/java/com/ovaplayer/data/model/VideoData.kt
package com.ovaplayer.data.model

data class VideoData(
    val videoId: String,
    val title: String,
    val filePath: String,
    val rating: Double,
    val durationSeconds: Int,
    val thumbnailPath: String?,  // nullable
    val previewPath: String?,    // nullable
    val tags: List<String>,
    val views: Int,
    val width: Int,
    val height: Int,
    val uploadedAt: String       // ISO 8601 timestamp as a string
)

File: core/frontend/src/app/data-types/video-data.ts
export interface VideoData {
  videoId: string;
  title: string;
  filePath: string;
  rating: number;
  durationSeconds: number;
  thumbnailPath: string;
  previewPath: string;
  tags: string[];
  views: number;
  width: number;
  height: number;
  uploadedAt: string; // ISO date string
}
